File: services/orchestration/models/events.py
"""AG-UI event models for streaming to the frontend."""

from __future__ import annotations
from typing import Any
from pydantic import BaseModel


class AGUIEvent(BaseModel):
    """An AG-UI protocol event sent via SSE."""
    type: str
    threadId: str = ""
    runId: str = ""
    messageId: str = ""
    role: str = ""
    delta: str = ""
    stepName: str = ""
    name: str = ""
    value: Any = None
    interrupts: list[dict[str, Any]] | None = None
    toolCallId: str = ""
    toolCallName: str = ""
    args: str = ""

    def to_sse(self) -> str:
        """Serialize to SSE data line."""
        import json
        payload = {k: v for k, v in self.model_dump().items() if v is not None and v != "" and v != []}
        payload["type"] = self.type
        return f"data: {json.dumps(payload)}\n\n"

File: apps/web/src/types/events.ts
/** AG-UI event types received from the backend SSE stream. */

export interface AGUIEvent {
  type: AGUIEventType;
  threadId?: string;
  runId?: string;
  messageId?: string;
  role?: string;
  delta?: string;
  stepName?: string;
  name?: string;
  value?: any;
  interrupts?: Array<{ id: string; value: any }>;
  toolCallId?: string;
  toolCallName?: string;
  args?: string;
}

export type AGUIEventType =
  | 'RUN_STARTED'
  | 'RUN_FINISHED'
  | 'RUN_ERROR'
  | 'STEP_STARTED'
  | 'STEP_FINISHED'
  | 'TEXT_MESSAGE_START'
  | 'TEXT_MESSAGE_CONTENT'
  | 'TEXT_MESSAGE_END'
  | 'TOOL_CALL_START'
  | 'TOOL_CALL_ARGS'
  | 'TOOL_CALL_END'
  | 'CUSTOM'
  | 'STATE_SNAPSHOT'
  | 'STATE_DELTA';

export interface ChatMessage {
  id: string;
  role: 'user' | 'assistant' | 'system';
  content: string;
  timestamp: string;
  agent?: string;
  isStreaming?: boolean;
}

export interface Scenario {
  id: string;
  title: string;
  description: string;
  icon: string;
  color: string;
  sample_prompts: string[];
}

export interface PromptCategory {
  name: string;
  icon: string;
  prompts: { text: string; scenario: string }[];
}
